tests/unit/session/test_session.py

```diff
@@ -285,6 +285,10 @@ def assert_popen_env(res):
         if tox_id != "GLOB":
             assert env["TOX_ENV_NAME"] == tox_id
             assert env["TOX_ENV_DIR"] == os.path.join(res.cwd, ".tox", tox_id)
+        # ensure native strings for environ for windows
+        for k, v in env.items():
+            assert type(k) is str, (k, v, type(k))
```

Comment:
@asottile 
A technical question: Why not use `isinstance(k, str)` instead `type`?

Reply:
`type(...)` is a stronger assertion here than `isinstance` -- I'm guarding against subclasses of `str`

Reply:
Thanks to the explanation.

```python
In [1]: class foostr(str):
   ...:     pass
   ...:
   ...:

In [2]: x = str("bar")

In [3]: y = foostr("bar")

In [4]: type(x)
Out[4]: str

In [5]: type(y)
Out[5]: __main__.foostr

In [6]: isinstance(x, str)
Out[6]: True

In [7]: isinstance(y, str)
Out[7]: True

In [8]: type(x) is str
Out[8]: True

In [9]: type(y) is str
Out[9]: False

```